sdk/keyvault/azure-security-keyvault-certificates/src/main/java/com/azure/security/keyvault/certificates/implementation/KeyVaultCredentialPolicy.java

```diff
@@ -235,7 +235,7 @@ public boolean authorizeRequestOnChallengeSync(HttpPipelineCallContext context,
         Optional<Object> contentLengthOptional = context.getData(KEY_VAULT_STASHED_CONTENT_LENGTH_KEY);
 
         if (request.getBody() == null && contentOptional.isPresent() && contentLengthOptional.isPresent()) {
-            request.setBody(BinaryData.fromObject(contentOptional.get()));
+            request.setBody((BinaryData) (contentOptional.get()));
```

Comment:
The CI pipeline tests have a few failures indicating we cannot cast things like `String` to `BinaryData`, is this expected?

Reply:
updated the tests